axelrod/match_generator.py

```diff
@@ -191,11 +217,7 @@ class SpatialMatches(RoundRobinMatches):
 
     def __init__(self, players, turns, game, repetitions, edges):
 
-        player_indices = list(range(len(players)))
-        node_indices = sorted(set([node for edge in edges for node in edge]))
-        if player_indices != node_indices:
-            raise ValueError("The graph edges do not include all players.")
-
+        test_graph_is_connected(edges, players)
```

Comment:
I agree. Just out with the little dude 🐶, will fix in a bit.

On Thu, 4 Aug 2016, 12:36 Owen Campbell, <notifications@github.com> wrote:

> In axelrod/match_generator.py
> <https://github.com/Axelrod-Python/Axelrod/pull/674#discussion_r73506231>:
>
> > @@ -191,11 +217,7 @@ class SpatialMatches(RoundRobinMatches):
> >
> >      def __init__(self, players, turns, game, repetitions, edges):
> >
> > -        player_indices = list(range(len(players)))
> > -        node_indices = sorted(set([node for edge in edges for node in edge]))
> > -        if player_indices != node_indices:
> > -            raise ValueError("The graph edges do not include all players.")
> > -
> > +        test_graph_is_connected(edges, players)
>
> I know that means a slight repeat of code, but it makes the function more
> readily understood, I reckon
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Axelrod-Python/Axelrod/pull/674/files/ff721b8c8ac18cce83a843405a3dbfc113ec9a13#r73506231>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACCGWvpu2qWO2YKeedDflXb-pDVO_jRJks5qcc7RgaJpZM4JZDqm>
> .
>